www/js/lib/uiUtil.js

```diff
@@ -45,35 +45,79 @@ define(rqDef, function(settingsStore) {
             // Display the modal
-            $("#alertModal").modal("show");
+            const modal = document.querySelector('#alertModal');
+            const backdrop = document.createElement('div');
+            backdrop.classList.add('modal-backdrop');
+            document.body.appendChild(backdrop);
+            // Set focus to the first focusable element inside the modal
+            modal.focus();
+
+            // Show the modal
+            document.body.classList.add('modal-open');
+            modal.classList.add('show');
+            modal.style.display = 'block';
+            backdrop.classList.add('show');
+
+            // Set the ARIA attributes for the modal
+            modal.setAttribute('aria-hidden', 'false');
+            modal.setAttribute('aria-modal', 'true');
+            modal.setAttribute('role', 'dialog');
+
+            // Hide modal handlers
+            function closeModalHandler(){
+              document.body.classList.remove('modal-open');
+              modal.classList.remove('show');
+              modal.style.display = 'none';
+              backdrop.classList.remove('show');
+              if(Array.from(document.body.children).indexOf(backdrop)>=0){ 
+                document.body.removeChild(backdrop);
+              }
+            }
             // When hide model is called, resolve promise with true if hidden using approve button, false otherwise
-            $("#alertModal").on("hide.bs.modal", function () {
-                const closeSource = document.activeElement;
-                if (closeSource.id === "approveConfirm") {
-                    resolve(true);
-                } else {
-                    resolve(false);
-                }
+            document.getElementById('modalCloseBtn').addEventListener('click', function close(){
+              closeModalHandler();
+              resolve(false);
             });
-            document.getElementById('alertModal').addEventListener('keyup', function (e) {
+            document.getElementById('declineConfirm').addEventListener('click', function () {
+              closeModalHandler();
+              resolve(false);
+            });
+            document.getElementById('closeMessage').addEventListener('click', function () {
+              closeModalHandler();
+              resolve(false);
+            });
+            document.getElementById('approveConfirm').addEventListener('click', function () {
+              closeModalHandler();
+              resolve(true);
+            });
+            
+            modal.addEventListener('click', function () {
+              closeModalHandler();
+              resolve(false);
+            });
+            document.getElementsByClassName('modal-dialog')[0].addEventListener('click', function(e){
+              e.stopPropagation();
+            })
+
+            document.addEventListener('keyup', function (e) {
```

Comment:
Sorry, I've found another issue while testing. If a document in the iframe is focused, then the Esc key doesn't register. To test, open a ZIM archive in the app and click on an external link in a document. This will open a dialogue box like below. Pressing Esc does nothing, but pressing Return works. However, if I first click on the dialogue box (focus it) before pressing Esc, then Esc does work.

To solve this, I think you have to focus the dialogue box programmatically when you show it. That might even allow you to add the event listeners on the modal rather than on document.

![image](https://user-images.githubusercontent.com/4304337/226618781-b16d7b2b-dfc8-45f7-9151-71145d09ecb1.png)

Reply:
Sorry, it was a mistake from my side, I did not notice it and it was working correctly in modal so it got missed

Reply:
Not a mistake, more a quirk in JS. Now that you've moved the focus, you should be able to add the keyup events to modal rather than to document.

However, event listeners are still getting defined multiple times and not removed. I think jQuery did that automatically. It might be necessary to add remove event listeners for all the listeners that are attached. This is less onerous than it looks because there is quite a lot of repeated code in the various ways of closing a dialogue. You can create a single function for this (`var dialogueClose = function () { ... };`) and then just remove it and attach it for each relevant event listener. You'll have to do a separate one for the confirm button. Plus the same for the key events as outlined above (but not attaching to document, attaching to modal).

Reply:
Have a look, I added some `removeEventListener`s. It should get fixed now

Reply:
@Jaifroid the commit I made before was not working, I eventually figured out how I should do it, it's working now. There are no event listeners remaining when one closes the Modal